sections/6-generic-programming.md

```diff
@@ -0,0 +1,63 @@
+## 6. Generic Programming
+
+<img src="https://raw.githubusercontent.com/monifu/scala-best-practices/master/assets/scala-logo-256.png"  align="right" width="128" height="128" />
+
+### 6.1. MUST NOT use automatic typeclass instance derivation
+
+When using [generic programming](https://github.com/milessabin/shapeless) or working with
```

Comment:
Would be nice to add also example how to do "derive encoder" in shapeless. If I am using circe, then example here is great, and I directly know what to do. If I would use shapeless, I would be slightly confused and have to go googling on how to do equivalent of deriveEncoder, or what is actually different.

Reply:
Hmmm. Yes not sure however whether it should go to best practices. Would a link to `circe` source code be enough (they have a really nice solution for it)?

Reply:
Link to example is good enough, as it's point to solution, instead of just saying this is bad.